src/mds/Server.cc

```diff
@@ -2531,7 +2531,39 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
     respond_to_request(mdr, mdr->more()->peer_error);
     return;
   }
-  
+
+  Session *session = mds->get_session(req);
+  if (session && !session->info.has_feature(CEPHFS_FEATURE_ALTERNATE_NAME) &&
+      req->get_op() != CEPH_MDS_OP_LOOKUP &&
+      req->get_op() != CEPH_MDS_OP_LOOKUPHASH &&
+      req->get_op() != CEPH_MDS_OP_LOOKUPPARENT &&
+      req->get_op() != CEPH_MDS_OP_LOOKUPINO &&
+      req->get_op() != CEPH_MDS_OP_LOOKUPNAME &&
+      req->get_op() != CEPH_MDS_OP_LOOKUPSNAP &&
+      req->get_op() != CEPH_MDS_OP_RMSNAP &&
+      req->get_op() != CEPH_MDS_OP_LSSNAP &&
+      req->get_op() != CEPH_MDS_OP_GETATTR &&
+      req->get_op() != CEPH_MDS_OP_READDIR &&
+      req->get_op() != CEPH_MDS_OP_UNLINK &&
+      req->get_op() != CEPH_MDS_OP_RMDIR) {
+    CInode *cur = try_get_auth_inode(mdr, req->get_filepath().get_ino());
```

Comment:
Here is no need to get the auth inode, the `acquire_locks()` will help you request the auth pin and auth locks from the auth MDS instead of forward the request directly to the auth MDS here.

Reply:
Ok, let me show my lack of knowledge on the MDS.  Should I simply replace that line with:

```
CInode *cur = mdcache->get_inode(req->get_filepath().get_ino());
```
or is this something more elaborated?

Reply:
IMO, this should be enough.